Repository/EntityRepositoryInterface.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+/*
+ * This file is part of the Doctrine Bundle
+ *
+ * The code was originally distributed inside the Symfony framework.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ * (c) Doctrine Project, Benjamin Eberlei <kontakt@beberlei.de>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Doctrine\Bundle\DoctrineBundle\Repository;
+
+use Doctrine\Common\Collections\Criteria;
+use Doctrine\Common\Persistence\ObjectRepository;
+use Doctrine\ORM\NativeQuery;
+use Doctrine\ORM\Query;
+use Doctrine\ORM\QueryBuilder;
+
+/**
+ * An interface that all entity repositories should implement.
+ *
+ * This contains all of the public methods found in Doctrine\ORM\EntityRepository
+ *
+ * @author Ryan Weaver <ryan@knpuniversity.com>
+ */
+interface EntityRepositoryInterface extends ObjectRepository
```

Comment:
I think this should be moved to Doctrine\ORM in the end. And let \Doctrine\ORM\EntityRepository implement it.

Reply:
I agree in principle. But I'm not sure when Doctrine 2.6 will be released, or if this could be put into that in time. I'd like to make this feature the default recommended way of doing things in Symfony 4 (and even update the generator to create these new service repositories to bump people in this direction).

There is a similar issue with the `RepositoryTrait` - it has copy-and-paste from `EntityRepository`... which in a more perfect world, could be replaced by a trait that lives in `Doctrine\ORM`. Of course, this could easily be done later without BC breaks.

So... in my perfect world, we would merge this PR, but also make these changes (add interface + the trait) in `Doctrine\ORM` and then use them later.

Reply:
This is pretty serious decision to introduce such interface in an integration bundle, which is IMHO not desired. It will become a huge pain due to BC when Doctrine 3 is released and/or when EntityRepositoryInterface is added to Doctrine itself.
I don't think it should exist here at all.